packages/code-du-travail-data/dump.js

```diff
@@ -35,20 +36,31 @@ const noIdSources = [
 ];
 
 async function fetchVector(data) {
-  return NLP_URL && data.title && data.text
-    ? vectorizeDocument(data.title, data.text)
+  if (NLP_URL) {
+    if (data.title && data.text) {
+      await vectorizeDocument(data.title, data.text)
         .then((title_vector) => {
           if (title_vector.message) {
             throw new Error(`error fetching ${data.title}`);
           }
           data.title_vector = title_vector;
-          return data;
         })
         .catch((err) => {
           console.error(`error fetching ${data.title}`, err.message);
-          return data;
-        })
-    : data;
+        });
+    } else {
+      // create random title_vector : search will fail if any of the docs
+      // does not have title_vector set : 512 dimension vector, 9 digits between -1 and 1
+      const length = 512;
+      const maxDigits = 9;
+      data.title_vector = Array.from(
+        { length },
+        () => (Math.random() > 0.5 ? 1 : -1) * Math.random().toFixed(maxDigits)
```

Comment:
I'm not sure that a random vector during indexation is advisable, if this vector is sneaky enough the related document might be in the result list of any query. It also introduce some randomness in the building process.
If the objective is solely to avoid errors, we could set an "extreme vector" for example a 512 zeros or ones vector. 
imho The best option would be to assign a "title" to these problematic document,  for example the body first sentence. @rmelisson Do you have an example of doc where the title is missing?

Reply:
So the 512 zeros option was responsible for NaN cosine similarity in Elastic, that's why I moved to randomness. 
We can indeed use the first sentence or any fake title. In any case, we won't solve the actual issue, which is to rely on a non-existing title. I agree with you about the randomness problem, we should at least have a consistent vector, I just thought random vector would be less prone to side effects than an "extreme vector".
Yep, I'll send some examples.

Reply:
can list the document that have no title ? if it's only version, then we should add a title / desc to it.
Keep in main that there is already some documents in the index that have a null vector but there a normally not searchable using vectors. Adding a "blank" vector avoid mapping error.

Reply:
it seems the documents with no title are now gone... at least if some appear again, we can deal with the case, and we log them